dietpi/dietpi-software

```diff
@@ -12701,16 +12701,20 @@ _EOF_
 			G_EXEC systemctl daemon-reload
 			G_EXEC systemctl restart docker
 
-			# Check for existing portainer installation
+			# Check for existing portainer data directory
 			if [[ -d '/mnt/dietpi_userdata/docker-data/volumes/portainer_data' ]]; then
 				G_DIETPI-NOTIFY 2 'Portainer detected, container will be dropped before reinstallation.'
-				G_EXEC docker rm "$(docker ps -a | mawk '/portainer\/portainer(-ce)?( |$)/{print $1}')" --force
-				G_EXEC docker rmi "$(docker images -a | mawk '/portainer\/portainer(-ce)?( |$)/{print $3}')"
 			else
 				# Create data directory
-				G_EXEC docker volume create portainer_data
+				docker volume ls -q | grep -q '^portainer_data$' || G_EXEC docker volume create portainer_data
```

Comment:
I think that above line would be fine without the if-then-else around. The above notification also could be confusing if the volume exists, but the container not, so no container is removed.

Reply:
ok I'm going to remove that park to keep it simple.